src/amo/components/AddonReview.js

```diff
@@ -0,0 +1,163 @@
+    this.state = { errorMessage: null };
+  }
+
+  onSubmit = (event) => {
+    event.preventDefault();
+    this.clearErrors();
+    const body = this.reviewTextarea.value;
+    if (!body) {
+      this.setState({
+        errorMessage: this.props.i18n.gettext(
+          'Please enter some text'
+        ),
+      });
+      return Promise.reject();
+    }
+    const params = {
+      body,
+      addonSlug: this.props.review.addonSlug,
+      reviewId: this.props.review.id,
+      apiState: this.props.apiState,
+      router: this.props.router,
+    };
+    // TODO: render a progress indicator in the UI.
+    return this.props.updateReviewText(params)
+      .then(() => this.goBackToAddonDetail());
+  }
+
+  goBackToAddonDetail = () => {
+    const { router } = this.props;
+    const { addonSlug } = this.props.review;
+    const { lang, clientApp } = this.props.apiState;
+    router.push(`/${lang}/${clientApp}/addon/${addonSlug}/`);
+  }
+
+  clearErrors() {
+    this.setState({ errorMessage: null });
+  }
+
+  render() {
+    const { i18n, review } = this.props;
+    if (!review || !review.id || !review.addonSlug) {
+      throw new Error(`Unexpected review property: ${JSON.stringify(review)}`);
+    }
+
+    let errorMessage;
+    if (this.state.errorMessage) {
+      errorMessage = (
+        <div className="AddonReview-error"
+             ref={(ref) => { this.errorMessage = ref; }}>
+          {this.state.errorMessage}
+        </div>
+      );
+    }
+
+    // TODO: I guess we should load the existing review text so it
+    // can be edited? That flow needs more thought.
+    return (
+      <div className="AddonReview">
+        <h2 className="AddonReview-header">{i18n.gettext('Write a review')}</h2>
+        <p>
+          {i18n.gettext(
+            'Tell the world why you think this extension is fantastic!'
+          )}
+        </p>
+        {errorMessage}
+        <form onSubmit={this.onSubmit} ref={(ref) => { this.reviewForm = ref; }}>
+          <textarea
+            className="AddonReview-textarea"
+            ref={(ref) => { this.reviewTextarea = ref; }}
+            name="review"
+            placeholder={i18n.gettext(
+              'Tell us what you love about this extension. ' +
+              'Be specific and concise.')} />
+          <div className="AddonReview-button-row">
+            <button className="AddonReview-button AddonReview-back-button"
+              onClick={this.goBackToAddonDetail}
+              ref={(ref) => { this.backButton = ref; }}>
+              {i18n.gettext('Back')}
+            </button>
+            <input
+              className="AddonReview-button AddonReview-submit-button"
+              type="submit" value={i18n.gettext('Submit')}
+            />
+          </div>
+        </form>
+      </div>
+    );
+  }
+}
+
+export const mapStateToProps = (state) => ({
+  apiState: state.api,
+});
+
+export const mapDispatchToProps = (/* dispatch */) => ({
+  updateReviewText(params) {
```

Comment:
well, I think one day soon it will need `dispatch()` when [callApi dispatches errors](https://github.com/mozilla/addons-frontend/issues/1032) but I can change it for now, that's fine.